### PR TITLE
Limit the number of devices we delete at once

### PR DESCRIPTION
**Pull request by erikjohnston**
Follow on from #14595

### CODE REVIEW

**Review by reivilibre — APPROVED** [review]

File: synapse/storage/databases/main/devices.py [hunk 1]
```diff
@@ -1614,20 +1618,21 @@ async def check_too_many_devices_for_user(self, user_id: str) -> List[str]:
 
         # Now fetch the devices to delete.
         sql = """
-            SELECT DISTINCT device_id FROM devices
+            SELECT device_id FROM devices
```
Comment (DMRobertson) [inline synapse/storage/databases/main/devices.py:1621]: Why no longer DISTINCT?

Comment (DMRobertson) [reply synapse/storage/databases/main/devices.py:1621]: oh, presumably (user_id, device_id) is already unique

Comment (erikjohnston) [reply synapse/storage/databases/main/devices.py:1621]: It was pointless mostly, as there are unique indexes on `user_id / device_id`

File: synapse/storage/databases/main/devices.py [hunk 2]
```diff
@@ -1614,20 +1618,21 @@ async def check_too_many_devices_for_user(self, user_id: str) -> List[str]:
 
         # Now fetch the devices to delete.
         sql = """
-            SELECT DISTINCT device_id FROM devices
```
Comment (MatMaul) [inline synapse/storage/databases/main/devices.py:1617]: FTR, `(device_id, user_id)` is unique by constraint hence the DISTINCT should indeed be useless.

**Review by MatMaul — APPROVED** [review]

**Pull request merged**